src/main/java/org/reficio/p2/FeatureBuilder.java

```diff
@@ -0,0 +1,226 @@
+		}
+		if (null != this.p2FeatureDefintion.getVersion()) {
+			featureElement.setAttribute("version", this.p2FeatureDefintion.getVersion());
+		} else if (featureElement.hasAttribute("version")) {
+			this.p2FeatureDefintion.setVersion(featureElement.getAttribute("version"));
+		} else {
+			throw new RuntimeException("No version defined for feature in pom or featureFile");
+		}
+		if (null != this.p2FeatureDefintion.getLabel()) {
+			featureElement.setAttribute("label", this.p2FeatureDefintion.getLabel());
+		}
+		if (null != this.p2FeatureDefintion.getProviderName()) {
+			featureElement.setAttribute("provider-name", this.p2FeatureDefintion.getProviderName());
+		}
+		if (null != this.p2FeatureDefintion.getDescription()) {
+			Element descriptionElement = XmlUtils.fetchOrCreateElement(xmlDoc, featureElement, "description");
+			descriptionElement.setTextContent(this.p2FeatureDefintion.getDescription());
+		}
+		if (null != this.p2FeatureDefintion.getCopyright()) {
+			Element copyrightElement = XmlUtils.fetchOrCreateElement(xmlDoc, featureElement, "copyright");
+			copyrightElement.setTextContent(this.p2FeatureDefintion.getCopyright());
+		}
+		if (null != this.p2FeatureDefintion.getLicense()) {
+			Element licenceElement = XmlUtils.fetchOrCreateElement(xmlDoc, featureElement, "license");
+			licenceElement.setTextContent(this.p2FeatureDefintion.getLicense());
+		}
+
+		// handle source feature renaming
+		if (generateSourceFeature) {
+			String id = featureElement.getAttribute("id");
+			if (!id.endsWith(".source")) {
+				// we assume, if ID does not end with .source, also label has no note that this is the developer resources
+				featureElement.setAttribute("id", id + ".source");
+				featureElement.setAttribute("label", featureElement.getAttribute("label") + " (Developer Resources)");
+			}
+		}
+		
+		for(P2Artifact artifact: this.p2FeatureDefintion.getArtifacts()) {
+			Collection<ArtifactBundlerInstructions> abis = this.bundlerInstructions.get(artifact);
+			for (ArtifactBundlerInstructions abi : abis) {
+				String id = abi.getSymbolicName();
+				String version = abi.getProposedVersion();
+				Element pluginElement = XmlUtils.createElement(xmlDoc,featureElement,"plugin");
+				pluginElement.setAttribute("id", id);
+				pluginElement.setAttribute("download-size", "0"); //TODO
+				pluginElement.setAttribute("install-size", "0");  //TODO
+				pluginElement.setAttribute("version", version);
+				pluginElement.setAttribute("unpack", "false"); // TODO
+				if (generateSourceFeature) {
+					id = abi.getSourceSymbolicName();
+					if (!StringUtils.isBlank(id)) {
+						pluginElement = XmlUtils.createElement(xmlDoc,featureElement,"plugin");
+						pluginElement.setAttribute("id", id);
+						pluginElement.setAttribute("download-size", "0"); //TODO
+						pluginElement.setAttribute("install-size", "0");  //TODO
+						pluginElement.setAttribute("version", version);
+						pluginElement.setAttribute("unpack", "false"); // TODO
+					}
```

Comment:
Will honor the unpack flag. I think I will need some help, how to determine the JAR size.
Also changed how the source plugin is generated.